sematic/scheduling/job_scheduler.py

```diff
@@ -20,151 +19,167 @@ class StateNotSchedulable(Exception):
-    field describing the new job details.
+    A new `Job` object will be appended to the end of the jobs for the
+    run describing the new job details.
 
     Parameters
     ----------
     run:
-        The run to schedule
+        The run to schedule.
     resolution:
-        The resolution associated with the run
+        The resolution associated with the run.
+    existing_jobs:
+        Any existent jobs associated with the run.
+
+    Returns
+    -------
+    A tuple where the first element is the run with its status updated and
+    the second element is the new list of jobs associated with the run.
     """
     # before scheduling a new run, update the information about previous runs
-    run.external_jobs = _refresh_external_jobs(run.external_jobs)
-    _assert_is_scheduleable(run, resolution)
-    external_jobs_list = list(run.external_jobs) + [_schedule_job(run, resolution)]
-    run.external_jobs = tuple(external_jobs_list)
+    existing_jobs = _refresh_jobs(existing_jobs)
+    _assert_is_scheduleable(run, resolution, existing_jobs)
+    jobs = list(existing_jobs) + [_schedule_job(run, resolution, existing_jobs)]
     run.future_state = FutureState.SCHEDULED
-    return run
+    logger.info("After schedule, jobs: %s", jobs)  # TODO: remove
+    return (run, jobs)
 
 
 def schedule_resolution(
     resolution: Resolution,
+    existing_jobs: Sequence[Job],
     max_parallelism: Optional[int] = None,
     rerun_from: Optional[str] = None,
-) -> Resolution:
+) -> Tuple[Resolution, Sequence[Job]]:
     """Start a resolution for the run on external compute.
 
     Parameters
     ----------
     resolution: Resolution
-        The resolution associated with the run
+        The resolution associated with the run.
+    existing_jobs:
+        Any existing jobs associated with the resolution.
     max_parallelism:
         The maximum number of non-inlined runs that the resolver will allow to be in the
         SCHEDULED state at any one time.
     rerun_from: Optional[str]
-        Start resolution from a particular point
+        Start resolution from a particular point.
+
+    Returns
+    -------
+    A tuple where the first element is the resolution with its status updated and
+    the second is the new list of jobs associated with the resolution.
     """
-    resolution.external_jobs = _refresh_external_jobs(resolution.external_jobs)
-    _assert_resolution_is_scheduleable(resolution)
-    external_jobs_list = list(resolution.external_jobs) + [
+    existing_jobs = _refresh_jobs(existing_jobs)
+    _assert_resolution_is_scheduleable(resolution, existing_jobs)
+    jobs = list(existing_jobs) + [
         _schedule_resolution_job(
             resolution=resolution,
             max_parallelism=max_parallelism,
             rerun_from=rerun_from,
         )
     ]
-    resolution.external_jobs = tuple(external_jobs_list)
     resolution.status = ResolutionStatus.SCHEDULED
-    return resolution
+    return (resolution, jobs)
```

Comment:
[Nit][Human linter] Are the parentheses necessary?

Reply:
I generally prefer to have them when returning tuples.